### ISSUE TITLE
Add jakarta classifier for rider-junit5 module

### ISSUE THREAD

**Issue by rmpestano** [database-rider/database-rider#565]
JUnit5 module still uses the default rider-core version (javax compatible), in order to use the Jakarta compatible version in JUnit5 module the user has to play with maven exclusions and explicitly add the rider-core with jakarta classifier dependency.

Provide Jakarta classifier for junit5 module so user can just declare rider-junit5:

```
     <dependency>
        <groupId>com.github.database-rider</groupId>
        <artifactId>rider-junit5</artifactId>
        <version>1.41.0</version>
        <scope>test</scope>
        <classifier>jakarta</classifier>
     </dependency>
```

**Comment by jonasrutishauser** [comment]
It seems that the rider-junit5 jakarta module is completely wrong:
- it still references the core module without the jakarta classifier.
- it still uses `javax.persistence`. For example in the `EntityManagerProvider`.

**Comment by rmpestano** [comment]
Hey @jonasrutishauser, EntityManagerProvider was not migrated to Jakarta, I'll reopen this issue.

Besides that you face other issues?

**Comment by jonasrutishauser** [comment]
At the moment, we explicitly reference the core with the jakarta qualifier and overwrite the `EntityManagerProvider` class with a `javax` to `jakarta` replaced variant. Additionally, we need a `Junit5RiderTestContext` compiled against the modified `EntityManagerProvider` to avoid a NoSuchMethodErrror. That's all to make our test running.

**Comment by jowko** [comment]
@rmpestano We tried to migrate our app to Jakarta API but we encountered issues with database-rider and junit5. We used classifiers `jakarta` for both `rider-core` and `rider-junit5` and used version `1.41.0` but we still encounter issues. 

DataSet does not work in our case anymore (we use H2 database for tests). When test is started, we get no data in DB. There is also no error logs about anything in our tests. I prepared very simple project where we updated Hibernate 5 from non Jakarta to Hibernate 5 with Jakarta and issue is reproducible there.

I also tried to rebuild rider libs from sources and I replaced all javax occurrences with jakarta but we still have the same issue. We also copied classes from `rider-junit5` into our project and applied fixes there but issue still occurs.

I analyzed code of your libraries and discovered, that at least  `rider-junit5 requires an update. For `rider-core`, you have `jakarta` profile which replaces some classes and imports to make code working on Jakarta. Such profile is present in `rider-junit5` but it lacks necessary configuration. Compare these two profiles:
https://github.com/database-rider/database-rider/blob/1.41.0/rider-core/pom.xml#L207
https://github.com/database-rider/database-rider/blob/1.41.0/rider-junit5/pom.xml#L117
In junit5, you only replace one dependency, but much more things needs to be replaced as in `rider-core` case.
I made such changes locally but we still have issues with test data not being inserted.

I will try to debug issue further but I do not have idea why it does not work.
I prepared repository where this issue can be easily reproduced: https://github.com/jowko/database-rider-bug

**Comment by jowko** [comment]
I believe that I found issue and it is present here where we get connection:
https://github.com/database-rider/database-rider/blob/master/rider-core/src/main/java/com/github/database/rider/core/util/EntityManagerProvider.jv#L124
This is invalid.

For Hibernate 5 with Jakarta, this is valid:
https://github.com/database-rider/database-rider/blob/master/rider-core/src/main/java/com/github/database/rider/core/util/EntityManagerProvider.java#L124

For Hibernate 6, such code is valid:
```
connection = em.unwrap( SessionImplementor.class )
                .getJdbcCoordinator()
                .getLogicalConnection()
                .getPhysicalConnection();
```
When I updated it, test data started to appear in database. I also updated example from https://github.com/jowko/database-rider-bug

**Comment by rmpestano** [comment]
Hey @jowko, thanks for your investigation. I'll try to fix that along the week and do a release, I let you know

**Comment by jowko** [comment]
@rmpestano Thanks for the response. If you prepare new version, I can test it right away to check if it is properly working.

**Comment by rmpestano** [comment]
Hi again @jowko, can you try with `1.41.1.SNAPSHOT` and classifier `jakarta`? you'll have to enable the snapshots repo on your maven settings like below:

```
<dependencies>
     <dependency>
        <groupId>com.github.database-rider</groupId>
        <artifactId>rider-junit5</artifactId>
        <version>1.41.1-SNAPSHOT</version>
        <scope>test</scope>
        <classifier>jakarta</classifier>
     </dependency>
</dependencies>

<repositories>
    <repository>
        <snapshots/>
        <id>snapshots</id>
        <name>libs-snapshot</name>
        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    </repository>
</repositories>
```

**Comment by jowko** [comment]
Hello @rmpestano. Thanks for the update. To make it work, I needed to use such config since `jakarta` variant of `rider-unit5` does not add `jakarta` classifier to `rider-core`:
```
<dependency>
    <groupId>com.github.database-rider</groupId>
    <artifactId>rider-junit5</artifactId>
    <version>1.41.1-SNAPSHOT</version>
    <scope>test</scope>
    <exclusions>
        <exclusion>
            <groupId>com.github.database-rider</groupId>
            <artifactId>rider-core</artifactId>
        </exclusion>
    </exclusions>
    <classifier>jakarta</classifier>
</dependency>
<dependency>
    <groupId>com.github.database-rider</groupId>
    <artifactId>rider-core</artifactId>
    <version>1.41.1-SNAPSHOT</version>
    <scope>test</scope>
    <classifier>jakarta</classifier>
</dependency>
```
I think that dependencies section from jakarta profile does not override dependencies in final pom: https://github.com/database-rider/database-rider/commit/c7cc010552cd5c090704e913eee1e92f9d1d348b#diff-2547568bf8c13ec17adbb35679a4fe48739d14433b1544482c49646f75211160L118

Also, `mvn dependency:tree` command returns such result if we declare just `rider-junit5`:
```
[INFO] +- com.github.database-rider:rider-junit5:jar:jakarta:1.41.1-SNAPSHOT:test
[INFO] |  \- com.github.database-rider:rider-core:jar:1.41.1-SNAPSHOT:test
```

From our side, current changes are sufficient but probably it would be better to address it to to avoid confusion for other users.
I also updated https://github.com/jowko/database-rider-bug so you can easily reproduce this issue if you comment out `rider-core` and exclusion from `rider-junit5` in pom.xml.

**Comment by rmpestano** [comment]
> I think that dependencies section from jakarta profile does not override dependencies in final pom

Yea, that's correct, when we deploy an artifact to maven central only one pom.xml is published and then a jar for each classifier (see latest rider release [here](https://repo1.maven.org/maven2/com/github/database-rider/rider-junit5/1.41.0/)) so I don't think we have much to do in this situation.

I'm thinking in releasing rider 2.0 without javax support, besides the issue you've described above  we also have #557 which is related to the javax/jakarta thingy and of course the whole complexity in the build it brings...anyway, I'm releasing 1.41.1 with the junit5 module fix and then prepare for 2.0. 

Thanks a lot for your feedback!

**Comment by jowko** [comment]
Thanks for the info. I will verify released version when it will be ready. I saw that some projects release jakarta/non jakarta version as separate artifacts, so this may be an option too (for example, `rider-core` and `rider-core-jakarta`). But releasing 2.0 would be much easier and probably more and more people will be using Jakarta only, since most major frameworks use it on it's newest versions.

**Comment by rmpestano** [comment]
The fix is available in `1.41.1` and as mentioned above, both `rider-core` and `rider-junit5` need to be declared with `jakarta` classifier
